### PR TITLE
[FEAT] Add rule no-object-literal-type-assertion

### PR DESCRIPTION
**Pull request by armano2**
This PR implements rule [no-object-literal-type-assertion](https://palantir.github.io/tslint/rules/no-object-literal-type-assertion/)

- #5

### CODE REVIEW

File: lib/rules/no-object-literal-type-assertion.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                "Forbids an object literal to appear in a type assertion expression",
+            extraDescription: [
+                util.tslintRule("no-object-literal-type-assertion"),
+            ],
+            category: "TypeScript",
+            url:
+                "https://github.com/nzakas/eslint-plugin-typescript/blob/master/docs/rules/no-object-literal-type-assertion.md",
+        },
+        messages: {
+            errorMessage:
```
Comment (j-f1) [inline lib/rules/no-object-literal-type-assertion.js]: How about `unexpectedTypeAssertion`?

File: docs/rules/no-object-literal-type-assertion.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Examples of **incorrect** code for this rule.
+
+```ts
+const x = { ... } as T;
+```
+
+Examples of **correct** code for this rule.
+
+```ts
+const x: T = { ... };
```
Comment (bradzacher) [inline docs/rules/no-object-literal-type-assertion.md]: looking at the tests - it's valid to cast to `any` or `unknown`.
Should probably document this here just to be clear.

Comment (armano2) [reply docs/rules/no-object-literal-type-assertion.md]: doc updated

**Comment by armano2** [comment]
looks like GH checks crashed, https://travis-ci.com/bradzacher/eslint-plugin-typescript/builds/94678838

**Pull request merged**